src/Core/Telemetry/TelemetryTracesHelper.cs

```diff
@@ -77,11 +77,11 @@ public static void TrackQueryActivityStarted(
         }
 
         /// <summary>
-        /// Tracks the completion of a REST controller activity.
+        /// Tracks the completion of a controller activity without any exceptions.
         /// </summary>
         /// <param name="activity">The activity instance.</param>
         /// <param name="statusCode">The HTTP status code of the response.</param>
-        public static void TrackRestControllerActivityFinished(
+        public static void TrackControllerActivityFinished(
```

Comment:
Does `TrackControllerActivityFinished` mark the finish of `TrackMainControllerActivityStarted` ? If yes, consider naming both start and finish symmetrically for easier interpretation.. 

so simply this `TrackControllerActivityStarted`  or add `Main` in the finished function